### PR TITLE
Add KCC Chain

### PR DESCRIPTION
**Pull request by enoughpower**
## Description

Add KCC Chain, closes https://github.com/trustwallet/wallet-core/issues/2064

KCC is a high-performance and decentralized public chain built by KCS’ and KuCoin’s fan communities.

Developed based on go-ethereum with the purpose of providing community users with a higher-speed, convenient and low-cost blockchain user experience.

KCC will have the following characteristics:

- Fully compatible with Ethereum and ERC-20 smart contracts, with extremely low migration costs;

- KCS will serve as the only core fuel and native token for KCC and can be used in scenarios such as gas fee payment;

- A block every 3 seconds results in faster transaction confirmation and higher chain performance;

- Proof of Staked Authority (PoSA) consensus algorithm, high efficiency, security and stability.

CoinMarketCap: https://coinmarketcap.com/currencies/kucoin-token/

## Testing instructions

Test in the same manner as BSC or Polygon was tested.

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [ ] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [ ] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

File: registry.json [hunk 1]
```diff
@@ -1918,5 +1918,29 @@
       "rpc": "https://smartbch.fountainhead.cash/mainnet",
       "documentation": "https://github.com/smartbch/docs/blob/main/developers-guide/jsonrpc.md"
     }
+  },
+  {
+    "id": "kcc",
+    "name": "KCC",
```
Comment (hewigovens) [inline registry.json:1924]: Is it the official name? not "Kucoin community Chain"?

File: docs/registry.md [hunk 2]
```diff
@@ -75,6 +75,7 @@ This list is generated from [./registry.json](../registry.json)
 | 10000118 | Osmosis          | OSMO   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/osmosis/info/logo.png" width="32" />      | <https://osmosis.zone/>       |
 | 10000145 | Smart Bitcoin Cash | BCH    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/smartbch/info/logo.png" width="32" />     | <https://smartbch.org/>       |
 | 10000250 | Fantom           | FTM    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/fantom/info/logo.png" width="32" />       | <https://fantom.foundation>   |
+| 10000321 | KCC              | KCS    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/kcc/info/logo.png" width="32" />          | <https://www.kcc.io/>         |
```
Comment (hewigovens) [inline docs/registry.md:78]: Let's add it to https://github.com/trustwallet/assets

File: registry.json (hunk 1 above)
Comment (enoughpower) [reply registry.json]: hello,we have change the official name with "Kucoin Community Chain"

File: docs/registry.md (hunk 2 above)
Comment (enoughpower) [reply docs/registry.md]: we will add logo and token assets to https://github.com/trustwallet/assets after this pr is merged,thanks!

**Review by pangsirirat422 — APPROVED** [review]

**Review by hewigovens — APPROVED** [review]

File: registry.json [hunk 3]
```diff
@@ -1918,5 +1918,29 @@
       "rpc": "https://smartbch.fountainhead.cash/mainnet",
       "documentation": "https://github.com/smartbch/docs/blob/main/developers-guide/jsonrpc.md"
     }
+  },
+  {
+    "id": "kcc",
+    "name": "KuCoin Community Chain",
+    "coinId": 10000321,
```
Comment (catenocrypt) [inline registry.json:1925]: I have an issue with coinId: we use the scheme of 10000000 + Id if two chains have the same coin (e.g. 714 for BNB on BC, 10000714 for BNB on BSC).
In this case 321 is not taken, so no justification for using 8-digit code.
Also, we always require the coinId to be listed in
https://github.com/satoshilabs/slips/blob/master/slip-0044.md
I don't see it there, can you add it, to make sure Id is not used by other chain?
321 is listed for Rapture RAP.

**Review by catenocrypt — CHANGES_REQUESTED** [review]

File: registry.json (hunk 3 above)
Comment (catenocrypt) [reply registry.json:1925]: OK, discussed this over.  Primary KCS is an ERC-20 token, so it does not have a Slip44 ID.
For KCC, coinID should be 10000000 + EthereumChainID, so 10000321 is fine.

**Review by catenocrypt — APPROVED** [review]

File: docs/registry.md (hunk 2 above)
Comment (enoughpower) [reply docs/registry.md]: https://github.com/trustwallet/assets/pull/19352 @catenocrypt  PR had created

File: docs/registry.md [hunk 4]
```diff
@@ -75,6 +75,7 @@ This list is generated from [./registry.json](../registry.json)
 | 10000118 | Osmosis          | OSMO   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/osmosis/info/logo.png" width="32" />      | <https://osmosis.zone/>       |
 | 10000145 | Smart Bitcoin Cash | BCH    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/smartbch/info/logo.png" width="32" />     | <https://smartbch.org/>       |
 | 10000250 | Fantom           | FTM    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/fantom/info/logo.png" width="32" />       | <https://fantom.foundation>   |
+| 10000321 | KuCoin Community Chain | KCS    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/kcc/info/logo.png" width="32" />          | <https://www.kcc.io/>         |
```
Comment (catenocrypt) [inline docs/registry.md:78]: Existence of logo in assets is prerequisite; please submit PR to assets repo adding logo, I will take care of it

**Review by catenocrypt — CHANGES_REQUESTED** [review]

File: docs/registry.md (hunk 4 above)
Comment (catenocrypt) [reply docs/registry.md:78]: Coin logo added via https://github.com/trustwallet/assets/pull/19362 (only coin logo so far, not the token ones).

**Review by catenocrypt — APPROVED** [review]

**Pull request merged**